### PR TITLE
Offchain worker: Enable http2 and improve logging

### PR DESCRIPTION
**Pull request by bkchr**
Apparently some webpages now return http2 by default and that silently breaks the offchain http
extension. The solution to this is to enable the `http2` feature of hyper. Besides that, this pr
improves the logging to make it easier to debug such errors.

Closes: https://github.com/paritytech/substrate/issues/10299

### CODE REVIEW

**Review by tomusdrw — APPROVED** [review]
Can we add a test that would query an external website to make sure it actually works?

**Comment by bkchr** [comment]
> Can we add a test that would query an external website to make sure it actually works?

I thought the same, but I'm not sure we should do this in CI. Querying something external means that our CI may breaks when the external thing is down or there was a flaky connection. TLDR. No fan of it. 

I mean I also wanted to have a test. I don't wanted to use hyper to send back a http2 answer, but maybe that isn't such a bad idea.

**Comment by tomusdrw** [comment]
> Querying something external means that our CI may breaks when the external thing is down or there was a flaky connection. TLDR. No fan of it.

Indeed, relying on external resources in unit tests is a controversial topic. I agree that we shouldn't be doing that in unit tests.

> I don't wanted to use hyper to send back a http2 answer, but maybe that isn't such a bad idea.

Yeah, sounds a bit complicated, but perhaps you're right. I think it can be just logged and worked on separately.

File: client/offchain/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,6 @@ hex = "0.4"
 fnv = "1.0.6"
 futures = "0.3.16"
 futures-timer = "3.0.2"
-log = "0.4.8"
```
Comment (kinosang) [inline client/offchain/Cargo.toml:22]: why move to `tracing` from `log`

Comment (bkchr) [reply client/offchain/Cargo.toml:22]: We already use tracing as the logging backend and the logging output looks much better. 

I especially like that you can pass "attributes" that are logged as well like `id` here:
```
tracing::debug!(target: LOG_TARGET, id = %request_id.0, "Finished writing body");
```

**Comment by bkchr** [comment]
bot merge

**Pull request merged**